### PR TITLE
Cpr 187 additional fields

### PR DESCRIPTION
**Pull request by vb-moj**
Refactored packages.

Refactored Offender Service.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/Aliases.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.hmpps.personrecord.client.model.offender
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import java.time.LocalDate
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class Aliases(
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/Aliases.kt:7]: Maybe should be named `Alias` as it only holds one alias?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.personrecord.client.model.offender
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class HighLight(
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt:7]: The case is a bit confusing - Highlight might be better but I suppose this is how the JSON is

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  @JsonProperty("surname")
+  val preferredName: List<String>? = emptyList(),
+) {
+  companion object
+  fun getPreferredNameString(): String? {
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt:12]: I don't really like the `String` on the end of this method name

**Review by markreesmoj — DISMISSED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/Aliases.kt (hunk 1 above)
Comment (vb-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/Aliases.kt:7]: Agree.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt (hunk 2 above)
Comment (vb-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt:7]: Probably a typo , will check and change.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt (hunk 3 above)
Comment (vb-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/offender/HighLight.kt:12]: Will check if Kotlin allows to override the getter.

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**